src/app/ClusterStateCache.cpp

```diff
@@ -66,14 +67,56 @@ CHIP_ERROR AttributeCache::UpdateCache(const ConcreteDataAttributePath & aPath,
     return CHIP_NO_ERROR;
 }
 
-void AttributeCache::OnReportBegin()
+CHIP_ERROR ClusterStateCache::UpdateEventCache(const EventHeader & aEventHeader, TLV::TLVReader * apData, const StatusIB * apStatus)
+{
+    if (apData)
+    {
+        //
+        // If we've already seen this event before, there's no more work to be done.
+        //
+        if (aEventHeader.mEventNumber < mHighestReceivedEventNumber)
```

Comment:
```suggestion
        if (aEventHeader.mEventNumber <= mHighestReceivedEventNumber)
```

Can we add a test for this?

Reply:
Honestly, not easily. FWIW, I don't think it's actually a big deal. This check here just optimizes doing redundant work, because without it, it just updates data for an existing event in the cache.

Testing for that isn't super easy without adding a lot of extra test-only hooks, so I'd rather just leave it un-tested for now to focus on other more important issues.

Reply:
So, this actually cannot be `<=`. That implies that at initialization time, we would have set it to 0. Which implies we've already received event 0, which isn't true.

In fact, the existing tests I added caught this, since I never saw event ID 0 coming through...

Reply:
I'm just gonna make it an Optional<EventNumber> since the semantic of `mHighestReceivedEventNumber` would be broken anyways without that.